jerry-ext/include/jerryscript-ext/arg.h

```diff
@@ -90,7 +90,61 @@ typedef enum
   JERRYX_ARG_REQUIRED
 } jerryx_arg_optional_t;
 
+/**
+ * Indicates the rounding policy which will be choose to transform a integer.
+ */
+typedef enum __attribute__((packed))
```

Comment:
the discussion with @martijnthe  is in this thread https://github.com/jerryscript-project/jerryscript/pull/1883#discussion_r120578346

The main purpose is that
* we don't want the signature of the jerryx_arg_[u]intX is too long
* jerryx_arg_int_option_t indicates that these are int-specific options from normal number. And people can just pass a empty one to use the default options
* we avoid the extra computation (like  encode (pack) the enums into extra_info) in jerryx_arg_[u]intX since they are inlined.